packages/commercetools/api-client/src/api/getProduct/index.ts

```diff
@@ -10,33 +8,26 @@ interface ProductData {
   products: ProductQueryResult;
 }
 
-const getProduct = async (search: ProductSearch): Promise<ApolloQueryResult<ProductData>> => {
-  if (search.customQuery) {
-    const { query, variables } = search.customQuery;
-
-    return await apolloClient.query<ProductData>({
-      query: gql`${query}`,
-      variables
-    });
-  }
-
+const getProduct = async (params, resolveQuery = async (query = defaultQuery, variables = {}) => {
```

Comment:
```suggestion
const getProduct = async (params, customQuery = async (query = defaultQuery, variables = {}) => {
```
according to the interface in the issue

Reply:
@filrak Actually this is more like resolving the final query then custom query for all cases. But we can leave it now, especially that we have multiple implementations now. 😃